### ISSUE TITLE
[test-triage] 33 failing tests in master not tagged as broken

### ISSUE THREAD

**Issue by drewmacrae** [lowRISC/opentitan#15877]
### Hierarchy of regression failure

Chip Level

### Failure Description

Ran: ```bazel test //... --test_tag_filters=-broken```
and found the following failures
```
//sw/device/tests:clkmgr_off_peri_test_sim_verilator                    TIMEOUT in 300.0s
//sw/device/tests/crypto:rsa_3072_verify_functest_wycheproof_fpga_cw310_test_rom TIMEOUT in 900.0s
//sw/device/tests/crypto:rsa_3072_verify_functest_wycheproof_sim_verilator TIMEOUT in 900.0s
//sw/device/examples/sram_program:sram_program_fpga_cw310_test_otp_dev   FAILED in 0.8s
//sw/device/examples/sram_program:sram_program_fpga_cw310_test_otp_rma   FAILED in 0.8s
//sw/device/examples/sram_program:sram_program_fpga_cw310_test_otp_test_unlocked0 FAILED in 8.0s
//sw/device/lib/crypto/drivers:entropy_test_sim_verilator                FAILED in 61.0s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_bad_identifier_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_bad_identifier_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_end_unaligned_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_end_unaligned_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_in_manifest_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_in_manifest_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_outside_image_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_outside_image_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_start_unaligned_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_start_unaligned_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_empty_code_a_sim_verilator FAILED in 60.1s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_empty_code_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_at_code_end_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_at_code_end_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_before_code_start_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_before_code_start_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_unaligned_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_unaligned_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_large_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_large_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_small_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_small_b_sim_verilator FAILED in 60.2s
//sw/device/tests:entropy_src_ast_rng_req_test_sim_verilator             FAILED in 87.1s
//sw/device/tests/crypto:aes_gcm_timing_test_fpga_cw310_rom              FAILED in 19.2s
//sw/device/tests/crypto:aes_gcm_timing_test_fpga_cw310_test_rom         FAILED in 18.2s
//sw/device/tests/crypto:aes_gcm_timing_test_sim_verilator               FAILED in 94.9s

Executed 890 out of 990 tests: 957 tests pass and 33 fail locally.
INFO: Build completed, 33 tests FAILED, 4771 total actions
```

### Steps to Reproduce

- Commit hash where failure was observed: 637313d14c387749321af3450b6c3d0c1dfbfc8a
Ran: ```bazel test //... --test_tag_filters=-broken```

but individual failures can likely be reproduced in isolations (except the timeouts that may be related to resource constraints)

**Comment by drewmacrae** [comment]
> //sw/device/tests/crypto:aes_gcm_timing_test_fpga_cw310_rom              FAILED in 19.2s
> //sw/device/tests/crypto:aes_gcm_timing_test_fpga_cw310_test_rom         FAILED in 18.2s
> //sw/device/tests/crypto:aes_gcm_timing_test_sim_verilator               FAILED in 94.9s

are a known issue addressed here https://github.com/lowRISC/opentitan/issues/15788
A resolution is planned for M3

**Comment by drewmacrae** [comment]
I've created an issue to track https://github.com/lowRISC/opentitan/issues/15934 for
//sw/device/tests:entropy_src_ast_rng_req_test_sim_verilator

**Comment by drewmacrae** [comment]
I'm not sure what in the driver should be sensitive to timing so I'm bisecting //sw/device/lib/crypto/drivers:entropy_test_sim_verilator to see if it's the icache again 🙃

**Comment by drewmacrae** [comment]
//sw/device/examples/sram_program:sram_program_fpga_cw310_test_otp_dev   FAILED in 0.8s
//sw/device/examples/sram_program:sram_program_fpga_cw310_test_otp_rma   FAILED in 0.8s
//sw/device/examples/sram_program:sram_program_fpga_cw310_test_otp_test_unlocked0 FAILED in 8.0s
are expected failures as I don't have a JTAG adapter connected. They should be tagged with JTAG by the test rule and filtered out

**Comment by drewmacrae** [comment]
I'm tracking //sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_bad_identifier_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_bad_identifier_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_end_unaligned_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_end_unaligned_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_in_manifest_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_in_manifest_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_outside_image_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_outside_image_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_start_unaligned_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_code_start_unaligned_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_empty_code_a_sim_verilator FAILED in 60.1s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_empty_code_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_at_code_end_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_at_code_end_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_before_code_start_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_before_code_start_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_unaligned_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_entry_unaligned_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_large_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_large_b_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_small_a_sim_verilator FAILED in 60.2s
//sw/device/silicon_creator/rom/e2e:boot_policy_bad_manifest_too_small_b_sim_verilator FAILED in 60.2s
with #16056

**Comment by drewmacrae** [comment]
There are now more failing tests, but here are the ones from this run not addressed above
```
//sw/device/tests:clkmgr_off_peri_test_sim_verilator                    TIMEOUT in 300.0s
//sw/device/tests/crypto:rsa_3072_verify_functest_wycheproof_fpga_cw310_test_rom TIMEOUT in 900.0s
//sw/device/tests/crypto:rsa_3072_verify_functest_wycheproof_sim_verilator TIMEOUT in 900.0s
//sw/device/lib/crypto/drivers:entropy_test_sim_verilator                FAILED in 61.0s
//sw/device/tests:entropy_src_ast_rng_req_test_sim_verilator             FAILED in 87.1s
```

**Comment by johngt** [comment]
Assuming that this is all M2 effort so labelled accordingly. If incorrect, please feel free to change the label @drewmacrae

**Comment by drewmacrae** [comment]
Looking into the wycheproof timeouts:

```
//sw/device/tests/crypto:rsa_3072_verify_functest_wycheproof_fpga_cw310_test_rom TIMEOUT in 900.0s
//sw/device/tests/crypto:rsa_3072_verify_functest_wycheproof_sim_verilator TIMEOUT in 900.0s
```
 Investigating for known good commit.
It's hanging on vector 170 (seemingly waiting for OTBN to complete) Instrumenting with logging info, has an effect on where it stalls so I should figure out halting and getting IP over GDB to see where it hangs when uninstrumented.

**Comment by drewmacrae** [comment]
Resolved by #16683
